tests/unit_tests/scheduler/test_scheduler.py

```diff
@@ -108,3 +107,16 @@ async def test_cancel(tmp_path: Path, realization):
 
     assert (tmp_path / "a").exists()
     assert not (tmp_path / "b").exists()
+
+
+async def test_that_max_submit_was_reached(tmp_path: Path, realization):
```

Comment:
should we ensure that the edge_case max_submit=1 is covered?

(what about max_submit=0, is that covered by the config parser?)

Reply:
Can do a parametriization ranging from 1 to 3? Regarding 0 that's a matter of config and not this test.